### ISSUE TITLE
[Question] How to view more of the conversation history

### ISSUE THREAD

**Issue by CSab6482** [sgtaziz/WebMessage#115]
![WebMessage Limitation](https://user-images.githubusercontent.com/53708281/110271708-92399000-7f96-11eb-964a-1dbb546a9405.JPG)
![PNG image](https://user-images.githubusercontent.com/53708281/110271719-936abd00-7f96-11eb-9c1d-1940c71fecc9.png)
I noticed that I can't see my entire message history in the WebMessage client. Is there a setting that I need to enable for this?

**Comment by sgtaziz** [comment]
Uhh. That's weird, as you scroll down it should keep loading more and more until there isn't any more data. I'll have to check

**Comment by CSab6482** [comment]
Okay, I have a little more info on this. It works as expected on my Mac client (but still stuck on the same 1/15/21 LS chat on Windows)
![Screen Shot 2021-03-08 at 12 14 16 PM](https://user-images.githubusercontent.com/53708281/110362040-ca7cb500-800e-11eb-828b-8c3e47642a00.png)
![PNG image](https://user-images.githubusercontent.com/53708281/110362048-cd77a580-800e-11eb-9842-a8246eed1108.png)
and on iOS I outlined the rough estimate of where contact images were in group chats if it would ever be possible to implement that.

![task manager](https://user-images.githubusercontent.com/53708281/110362098-dff1df00-800e-11eb-8589-97ccba7e3260.JPG)
However, I scrolled up about 12 hours worth of messages in one chat to find something and then it started going really slow (scrolling, typing, everything)
Until it crashed, and upon going back into my WebMessage client, I couldn't open any messages
![doesn't open](https://user-images.githubusercontent.com/53708281/110362204-fdbf4400-800e-11eb-8904-7d5f5d5f190b.JPG)
[similar crash.txt](https://github.com/sgtaziz/WebMessage/files/6103266/similar.crash.txt)
^ this is the crash log for that specific incident

[crash transcoder.txt](https://github.com/sgtaziz/WebMessage/files/6103268/crash.transcoder.txt)
^and here is the crash log for a similar incident

Let me know if I should test anything further.

**Comment by CSab6482** [comment]
[jetsam memory crash.txt](https://github.com/sgtaziz/WebMessage/files/6103642/jetsam.memory.crash.txt)
I have another jetsam memory crash here

**Comment by CSab6482** [comment]
Hmm, it looks like the jetsam crashes [are normal](https://www.reddit.com/r/jailbreak/comments/m34qq2/question_jetsam_memory_limits_with_odysseyra1n_on/gqmwpyi?utm_source=share&utm_medium=web2x&context=3)

**Comment by sgtaziz** [comment]
Jetsam crashes are indeed normal most of the times, I override the memory limit manually. However, still, I can't give it infinite memory. If you are crashing towards the end of your conversations, I would assume that you're running out of memory on your device. Which actually should not happen unless you have a weird/buggy thread there? I can personally scroll throughout all my history (until 2017) on my macOS and Windows machines

**Comment by CSab6482** [comment]
Okay, so now on Windows, I can only scroll as far back as 1/18/21 in the left-hand side chat history. However, if I begin a "new" chat with someone that I have a history with before that date, my previous messages with them will show up in the compose new message Window. As for scrolling up in a chat, I remember that the particular chat I was scrolling up in then had a lot of attachments, so maybe that had something to do with it. I just tested it in another chat, and I was able to scroll without any issues for the most part, however, some messages did duplicate.
![duplicate](https://user-images.githubusercontent.com/53708281/111037355-a8f64180-83f1-11eb-8296-19be8f1f5dce.png)
![duplicate 2](https://user-images.githubusercontent.com/53708281/111037366-b0b5e600-83f1-11eb-8946-914584b60f18.png)
These messages were messages that were at the top as more message history was loading.

**Comment by sgtaziz** [comment]
For the lag, did you update the tweak? I pushed some changes that should help with that as well

**Comment by CSab6482** [comment]
Yes, I updated both the tweak and the client, and for general use, the lag is much better. This was just one instance in which I loaded lots of old messages in one chat, so then that one chat became laggy. I'll test later if chats that have many incoming and outgoing messages in a short amount of time are still affected.

**Comment by shallax** [comment]
Same problem - happens when scrolling up and down the list of chats rather than a single chat.

**Comment by sgtaziz** [comment]
Any info on this with 0.7.0?

**Comment by CSab6482** [comment]
On version 0.7.0, I can still only see fifty chats in my conversation history, this time going as far back as 2/26/21
